### DISCUSSION TITLE
Spoiler component: allow ellipsis when text is truncated

### DISCUSSION THREAD

**Discussion by pauloendoh** [mantinedev/mantine#6488] [Feature requests]
The `<Spoiler>` component does not show ellipsis ("...") at the end of the text when text is truncated. 

![image](https://github.com/mantinedev/mantine/assets/17272651/a97a8074-6d07-47df-9784-ef156c0a32c7)
(example from Mantine docs)

It would be nice to have a prop to indicate what should be used when the text is truncated.

Eg: `<Spoiler  truncatedEndingComponent={"... "} />`

**Comment by rtivital** [comment]
Not planned to be implemented as part of the Spoiler component, this feature is available in Text component – https://mantine.dev/core/text/#line-clamp

**Reply by AlonMiz** [reply]
this should probably be an ability of the Text (truncate/line clamp) + tooltip
beacuse achieveing that manually would require to get hold of the ref and calculate that yourself. and it's very verbose.
i would imagine something like this

```
<Text truncate truncateTooltip={TooltipPropsWithoutLabel} />
```

**Comment by puyanwei** [comment]
This does not work together, it just shows the whole text without any show more/show less button. 

Seems like quite a common combination used on many websites, would love to see it implemented!
```
    const [expanded, setExpanded] = useState(false);

  <Spoiler expanded={expanded} onExpandedChange={() => setExpanded(!expanded)}>
      <Text lineClamp={2}>{text}</Text>
  </Spoiler>
```

**Reply by puyanwei** [reply]
My solution is to tie the Text clamp with the expanded boolean. Note the padding top needed as for some reason the description still shows up slightly. Hopefully this helps others who want this feature!

```
<Spoiler
  expanded={expanded}
  hideLabel={hideLabel}
  showLabel={showLabel}
  onExpandedChange={setExpanded}
>
  <Text className="pt-1" lineClamp={expanded ? 0 : 3}>
    {description}
  </Text>
</Spoiler>
```

**Reply by kalnode** [reply]
This technique works but it seems to cancel-out the spoiler's close animation when hiding the content. At this point aren't you just halfway to making a new custom component though.

In any case, I agree that spoiler-with-text seems isn't that great without ellipsis.

The other thing commonly done is a fade-out, but usually that's done with a gradient overlay to a solid colour that matches the page background.

e.g. just throwing this in your CSS will get you a fade to work from:

```
/* Adds bottom fade-out over content */
.mantine-Spoiler-content::after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translate(-50%, 0);
    min-height: 70%;
    width: 100%;
    background: linear-gradient(180deg, rgba(255, 255, 255, 0) 10%, rgb(255, 255, 255) 130%);
    pointer-events: none;
}

/* Hides fade-out when spoiler is expanded */
.mantine-Spoiler-control[aria-expanded=true] + .mantine-Spoiler-content::after { 
    display: none;
}

```
